File: 프로그래머스/unrated/181882. 조건에 맞게 수열 변환하기 1/조건에 맞게 수열 변환하기 1.js
function solution(arr) {
  let ans = []

  for (let i = 0; i < arr.length; i++) {
    if (arr[i] >= 50 && arr[i] % 2 == 0) {
      ans.push(arr[i] / 2)
    } else if (arr[i] < 50 && arr[i] % 2 == 0) {
      ans.push(arr[i])
    } else if (arr[i] > 50 && arr[i] % 2 == 1) {
      ans.push(arr[i])
    } else {
      ans.push(arr[i] * 2)
    }
  }
  return ans
}

File: 프로그래머스/0/181882. 조건에 맞게 수열 변환하기 1/조건에 맞게 수열 변환하기 1.java
class Solution {
  public int[] solution(int[] arr) {
    int[] result = new int[arr.length];

    for (int i = 0; i < arr.length; i++) {
      if (arr[i] >= 50 && arr[i] % 2 == 0) result[i] = arr[i] / 2;
      else if (arr[i] < 50 && arr[i] % 2 == 1) result[i] = arr[i] * 2;
      else result[i] = arr[i];
    }

    return result;
  }
}